Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/ProjectionIntegrityViolationDetector.php

```diff
@@ -175,74 +175,29 @@ public function tetheredNodesAreNamed(): Result
     /**
      * @inheritDoc
      */
-    public function restrictionsArePropagatedRecursively(): Result
+    public function subtreeTagsAreInherited(): Result
     {
         $result = new Result();
-        $nodeRecordsWithMissingRestrictions = $this->client->getConnection()->executeQuery(
-            'SELECT c.nodeaggregateid, h.contentstreamid, h.dimensionspacepoint
-            FROM ' . $this->tableNamePrefix . '_hierarchyrelation h
-            INNER JOIN ' . $this->tableNamePrefix . '_node p
-                ON p.relationanchorpoint = h.parentnodeanchor
-            INNER JOIN ' . $this->tableNamePrefix . '_restrictionrelation pr
-                ON pr.affectednodeaggregateid = p.nodeaggregateid
-                AND pr.contentstreamid = h.contentstreamid
-                AND pr.dimensionspacepointhash = h.dimensionspacepointhash
-            INNER JOIN ' . $this->tableNamePrefix . '_node c
-                ON c.relationanchorpoint = h.childnodeanchor
-            LEFT JOIN ' . $this->tableNamePrefix . '_restrictionrelation cr
-                ON cr.affectednodeaggregateid = c.nodeaggregateid
-                AND cr.contentstreamid = h.contentstreamid
-                AND cr.dimensionspacepointhash = h.dimensionspacepointhash
-            WHERE cr.affectednodeaggregateid IS NULL'
+        $hierarchyRelationsWithMissingSubtreeTags = $this->client->getConnection()->executeQuery(
+            'SELECT
+              ph.name
+            FROM
+              ' . $this->tableNamePrefix . '_hierarchyrelation h
+              INNER JOIN ' . $this->tableNamePrefix . '_hierarchyrelation ph
+                ON ph.childnodeanchor = h.parentnodeanchor
+                AND ph.contentstreamid = h.contentstreamid
+                AND ph.dimensionspacepointhash = h.dimensionspacepointhash
+            WHERE
+              EXISTS (
+                SELECT t.tag FROM JSON_TABLE(JSON_KEYS(ph.subtreetags), \'$[*]\' COLUMNS(tag VARCHAR(30) PATH \'$\')) t WHERE NOT JSON_EXISTS(h.subtreetags, CONCAT(\'$.\', t.tag))
+              )'
```

Comment:
This part determines if a parent hierarchy relation contains subtree tags that are not existing in the child relation.
This could probably be solved with `JSON_ARRAY_INTERSECT(JSON_KEYS(ph.subtreetags), JSON_KEYS(h.subtreetags)` but unfortunately that's only available with MariaDB 11.2+ according to https://mariadb.com/kb/en/json_array_intersect/

Reply:
todo move comment to code